core/capabilities/ccip/ccipevm/helpers_test.go

```diff
@@ -38,4 +39,57 @@ func Test_decodeExtraArgs(t *testing.T) {
 
 		require.Equal(t, gasLimit, decodedGasLimit)
 	})
+
+	t.Run("decode extra args into map evm v1", func(t *testing.T) {
+		encoded, err := d.contract.EncodeEVMExtraArgsV1(nil, message_hasher.ClientEVMExtraArgsV1{
+			GasLimit: gasLimit,
+		})
+		require.NoError(t, err)
+
+		m, err := DecodeExtraArgsToMap(encoded)
+		require.NoError(t, err)
+		require.Len(t, m, 1)
+
+		gl, exist := m["gasLimit"]
+		require.True(t, exist)
+		require.Equal(t, gl, gasLimit)
+	})
+
+	t.Run("decode extra args into map evm v2", func(t *testing.T) {
+		encoded, err := d.contract.EncodeEVMExtraArgsV2(nil, message_hasher.ClientEVMExtraArgsV2{
+			GasLimit:                 gasLimit,
+			AllowOutOfOrderExecution: true,
+		})
+		require.NoError(t, err)
+
+		m, err := DecodeExtraArgsToMap(encoded)
+		require.NoError(t, err)
+		require.Len(t, m, 2)
+
+		gl, exist := m["gasLimit"]
+		require.True(t, exist)
+		require.Equal(t, gl, gasLimit)
+
+		ooe, exist := m["allowOutOfOrderExecution"]
+		require.True(t, exist)
+		require.Equal(t, true, ooe)
+	})
+
+	t.Run("decode extra args into map svm", func(t *testing.T) {
+		key, err := solana.NewRandomPrivateKey()
+		require.NoError(t, err)
+		encoded, err := d.contract.EncodeSVMExtraArgsV1(nil, message_hasher.ClientSVMExtraArgsV1{
+			ComputeUnits:             10000,
+			AccountIsWritableBitmap:  4,
+			AllowOutOfOrderExecution: false,
+			TokenReceiver:            [32]byte(key.PublicKey().Bytes()),
+			Accounts: [][32]byte{
+				[32]byte(key.PublicKey().Bytes()),
+			},
+		})
+		require.NoError(t, err)
+
+		_, err = DecodeExtraArgsToMap(encoded)
+		require.NoError(t, err)
```

Comment:
once the test passes, we should verify the decoded data is same as originally passed in data.